prometheus-exporter/src/main/java/be/xplore/githubmetrics/prometheusexporter/MetricScheduler.java

```diff
@@ -0,0 +1,81 @@
+package be.xplore.githubmetrics.prometheusexporter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import org.springframework.scheduling.support.CronTrigger;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+
+@ConditionalOnProperty(
+        value = "app.scheduling.enable", havingValue = "true", matchIfMissing = true
+)
+@Service
+@EnableScheduling
```

Comment:
Gerd was talking yesterday that this should live on a configuration class?

Reply:
top, done! Put in its own class that also has the `@ConditionalOnProperty` so that its only active if scheduling is active